cscs-checks/prgenv/mpi.py

```diff
@@ -0,0 +1,46 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.required_version('>=2.14')
+@rfm.parameterized_test(['single'], ['funneled'], ['serialized'], ['multiple'])
+class Mpi_InitTest(rfm.RegressionTest):
+    def __init__(self, required_thread):
+        super().__init__()
+        self.descr = 'MPI_Init_thread ' + required_thread
+        self.valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
+                                    'PrgEnv-intel', 'PrgEnv-pgi']
+        self.build_system = 'SingleSource'
+        self.sourcepath = 'mpi_init_thread.cpp'
+        self.cppflags = {
+            'single':     ['-D_MPI_THREAD_SINGLE'],
+            'funneled':   ['-D_MPI_THREAD_FUNNELED'],
+            'serialized': ['-D_MPI_THREAD_SERIALIZED'],
+            'multiple':   ['-D_MPI_THREAD_MULTIPLE']
+        }
+        self.build_system.cppflags = self.cppflags[required_thread] + ['-static']
```

Comment:
Split in two lines

Reply:
3122756caa398ff5c756822db5e54276c0e8bae1